sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs

```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
     /// Gets the format to read and write the model.
     /// </summary>
     public string Format { get; }
+
+    /// <summary>
+    /// Registers an <see cref="IPersistableModel{T}"/> proxy to be used when reading or writing a model.
+    /// </summary>
+    /// <param name="proxy"> The <see cref="IPersistableModel{T}"/> proxy that will be used to read or write the model. </param>
+    public void AddProxy<T>(IPersistableModel<T> proxy)
+    {
+        if (_proxies is null)
+            _proxies = new Dictionary<Type, object>();
+
+        _proxies.Add(typeof(T), proxy);
+    }
+
+    /// <summary>
+    /// Gets the model that is currently being proxied.
+    /// </summary>
+    /// <remarks>
+    /// Whenever a proxy is used for a given type, the model being proxied is set here.
+    /// The value does not need to be cleared after setting it will be overwritten on the next use.
```

Comment:
Does this line (59) need to be present if this property is no longer settable by the end-user?